### PR TITLE
Simplified the form theming docs

### PR DESCRIPTION
**Pull request by javiereguiluz**
This is a total revamp of all the docs related to form themes and form customization. The current problems are:

* `form/form_customization.rst` and `form/rendering.rst` are too similar.
* `form/form_themes.rst` is incomplete and confusing
* Most of the form theming things are explained in `form/form_customization.rst`
  instead of `form/form_themes.rst`
* `reference/forms/twig_reference.rst` and `reference/twig_reference.rst`
  duplicate lots of contents

This PR does the following:

* Delete `form/rendering.rst` and merge it with `form/form_customization.rst`
* Remove anything related to form themes from `form/form_customization.rst`
* Add everything related to form themes to ``form/form_themes.rst``
* Remove the duplicated contents from `reference/twig_reference.rst` and link
  to the other form reference.

The result is:

* `form.rst`: main article explaining all the basics about forms.
* `form/form_customization.rst` explains everything about Form helper functions
  (`form_start()`, `form_row()`, etc.) and form variables (`required`, `compound`, etc.)
* `form/form_themes.rst` explains everything about form themes, including the
  form fragment naming and all the different options to create themes.

On top of all the above, this PR removes 800 lines of docs!

### CODE REVIEW

File: form/form_themes.rst [hunk 1]
```diff
@@ -5,185 +5,372 @@
...
-and different customizations can be used in different places.
+This article explains how to use in your app any of the form themes provided by
+Symfony and how to create your own custom form theme.
+
+.. _symfony-builtin-forms:
+
+Symfony Built-In Form Themes
+----------------------------
+
+Symfony comes with several **built-in form themes** that makes your forms look
```
Comment (atailouloute) [inline form/form_themes.rst]: I think it should be `make` not `makes`, form themes are in plural form

File: reference/configuration/twig.rst [hunk 2]
```diff
@@ -196,6 +197,62 @@ option is advanced. If you need to customize an error page you should use
...
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:twig="http://symfony.com/schema/dic/twig"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/twig http://symfony.com/schema/dic/twig/twig-1.0.xsd">
+
+            <twig:config>
+                <twig:theme>bootstrap_4_layout.html.twig</twig:theme>
```
[the remark this answers is not in the document]
Comment (javiereguiluz) [reply reference/configuration/twig.rst]: I don't know. All docs used `<twig:theme>` before, so I kept it. Let's ask @xabbuh.

[the remark this answers is not in the document]
Comment (xabbuh) [reply reference/configuration/twig.rst]: I don't think `<twig:theme>` will work. `<twig:form-theme>` looks correct to me.

**Comment by javiereguiluz** [comment]
I'm going to merge this because it's hard to review using GitHub's interface. We can create new PRs to fix the things we don't like after the merge. Thanks!

**Pull request closed without merging**